lrpar/src/lib/lex.rs

```diff
@@ -143,3 +143,11 @@ impl<StorageT: Copy> Lexeme<StorageT> {
         self.len == u32::max_value()
     }
 }
+
+impl<StorageT: Copy> fmt::Display for Lexeme<StorageT> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Lexeme[{}..{}]", self.span().start(), self.span().end())
```

Comment:
Unfortunately, neither the lexeme nor the span has a reference to the input so it can't print out anything to do with the actual input.

However, an interesting question is whether we should also print out the lexeme type and whether it's been inserted by error correction or not. In a sense, I almost think that `Display` on `Lexeme` might as well be equivalent to `Debug`. Thoughts?

Reply:
Ah, right! Missed that we don't have the input here.

>  In a sense, I almost think that Display on Lexeme might as well be equivalent to Debug.

I think that will be a bit verbose. `Lexeme[12..24]` looks good to me.

> However, an interesting question is whether we should also print out the lexeme type and whether it's been inserted by error correction or not.

Would be interesting to include. Do you have an example how it would look like?

Reply:
If you do `println!("{:?}", $1)` or similar in action code you should see what the `Debug` output looks like currently.

Reply:
In retrospect, I'm inclined to say that what this PR proposes for `Display` on a `Lexeme` is probably a reasonable first start: we can adjust as we go forward, and if people want more information (lexeme type, is/is-not inserted) they can fairly easily write their own pretty printer.